pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/DSMServer.java

```diff
@@ -696,71 +710,34 @@ private void setupCohortTagRoutes() {
 
     private void setupPubSub(@NonNull Config cfg, NotificationUtil notificationUtil) {
         String projectId = cfg.getString(GCP_PATH_PUBSUB_PROJECT_ID);
-        String subscriptionId = cfg.getString(GCP_PATH_PUBSUB_SUB);
         String dsmToDssSubscriptionId = cfg.getString(GCP_PATH_DSM_DSS_SUB);
         String dsmTasksSubscriptionId = cfg.getString(GCP_PATH_DSM_TASKS_SUB);
         String mercuryDsmSubscriptionId = cfg.getString(GCP_PATH_MERCURY_DSM_SUB);
         String antivirusDsmSubscriptionId = cfg.getString(GCP_PATH_ANTI_VIRUS_SUB);
 
-        logger.info("Setting up pubsub for {}/{}", projectId, subscriptionId);
-
-        try {
-            // Instantiate an asynchronous message receiver.
-            MessageReceiver receiver = (PubsubMessage message, AckReplyConsumer consumer) -> {
-                // Handle incoming message, then ack the received message.
-                try {
-                    TransactionWrapper.inTransaction(conn -> {
-                        PubSubLookUp.processCovidTestResults(conn, message, notificationUtil);
-                        logger.info("Processing the message finished");
-                        consumer.ack();
-                        return null;
-                    });
-
-                } catch (Exception ex) {
-                    logger.info("about to nack the message", ex);
-                    consumer.nack();
-                    ex.printStackTrace();
-                }
-            };
-
-            Subscriber subscriber = null;
-            ProjectSubscriptionName resultSubName = ProjectSubscriptionName.of(projectId, subscriptionId);
-            ExecutorProvider resultsSubExecProvider = InstantiatingExecutorProvider.newBuilder().setExecutorThreadCount(1).build();
-            subscriber = Subscriber.newBuilder(resultSubName, receiver).setParallelPullCount(1).setExecutorProvider(resultsSubExecProvider)
-                    .setMaxAckExtensionPeriod(org.threeten.bp.Duration.ofSeconds(120)).build();
-            try {
-                subscriber.startAsync().awaitRunning(1L, TimeUnit.MINUTES);
-                logger.info("Started pubsub subscription receiver for {}", subscriptionId);
-            } catch (TimeoutException e) {
-                throw new DsmInternalError("Timed out while starting pubsub subscription " + subscriptionId, e);
-            }
-        } catch (Exception e) {
-            throw new DsmInternalError("Failed to get results from pubsub ", e);
-        }
-
         logger.info("Setting up pubsub for {}/{}", projectId, dsmToDssSubscriptionId);
 
         try {
-            PubSubResultMessageSubscription.dssToDsmSubscriber(projectId, dsmToDssSubscriptionId);
+            dssSubsciber = PubSubResultMessageSubscription.dssToDsmSubscriber(projectId, dsmToDssSubscriptionId);
```

Comment:
For proper shutdown, it was necessary to track the subscribers explicitly so they could be shutdown explicitly.